gzguts.h

```diff
@@ -139,6 +139,7 @@ void Z_INTERNAL gz_error(gz_state *, int, const char *);
 /* GT_OFF(x), where x is an unsigned value, is true if x > maximum z_off64_t
    value -- needed when comparing unsigned to z_off64_t, which is signed
    (possible z_off64_t types off_t, off64_t, and long are all signed) */
+unsigned Z_INTERNAL gz_intmax(void);
```

Comment:
This commit is not required.
``gz_intmax`` is not exported in zlib (``gz_intmax`` in ``local`` section of https://github.com/madler/zlib/blob/develop/zlib.map).
In zlib-ng it is removed from the ``local`` section of ``zlib.map`` and ``zlib-ng.map`` (commit: https://github.com/zlib-ng/zlib-ng/commit/b1ebcc0ee988b398580bced4a90050d41ec38133).
It's does not affect the ABI.

Reply:
But wouldn't it still be necessary for ZLIB_COMPAT when compiling statically?

Reply:
Interesting question!


If we need to maintain compatibility of internal (non exported) interfaces for static library then I suggest making this change like:

```c
#ifdef ZLIB_COMPAT
unsigned Z_INTERNAL gz_intmax(void);
#endif
```

```c
#ifdef ZLIB_COMPAT

/* portably return maximum value for an int (when limits.h presumed not
   available) -- we need to do this to cover cases where 2's complement not
   used, since C standard permits 1's complement and sign-bit representations,
   otherwise we could just use ((unsigned)-1) >> 1 */
unsigned Z_INTERNAL gz_intmax(void) {
    return INT_MAX;
}

#endif
```

Or add the special ``ZLIB_COMPAT_STATIC`` macros only for a ``ZLIB_COMPAT``-static library target.

Reply:
Fixed.